### PR TITLE
Many logging fixes to quiet things down and make errors more useful.

### PR DESCRIPTION
**Pull request by spencerkimball**
Set default range retry options (they were never set after a prior change).

If a leader returned via NotLeaderError is not one of the known replicas in a range
descriptor, the dist sender will evict the entry from the range descriptor cache.

### CODE REVIEW

File: proto/config.go [hunk 1]
```diff
@@ -179,6 +179,11 @@ func (p *PermConfig) CanWrite(user string) bool {
 	return false
 }
 
+// Stringer implementation for replica.
+func (r *Replica) String() string {
+	return fmt.Sprintf("%d:%d", r.NodeID, r.StoreID)
+}
+
 // A ReplicaSlice is a slice of Replicas.
 type ReplicaSlice []Replica
```
Comment (tschottdorf) [inline proto/config.go]: Can you rebase? This collides with @petermattis recently merged change.

File: util/uuid.go [hunk 2]
```diff
@@ -57,3 +57,11 @@ func (u UUID) String() string {
 	b := []byte(u)
 	return fmt.Sprintf("%08x-%04x-%04x-%04x-%012x", b[:4], b[4:6], b[6:8], b[8:10], b[10:])
 }
+
+func (u UUID) Short() string {
+	if u == nil || len(u) != UUIDSize {
```
Comment (tschottdorf) [inline util/uuid.go]: the latter implies the first.

**Comment by tschottdorf** [comment]
LGTM mod it needing a rebase.

File: util/uuid.go (hunk 2 above)
Comment (spencerkimball) [inline util/uuid.go]: Done.

File: proto/config.go (hunk 1 above)
Comment (spencerkimball) [inline proto/config.go]: Yup.

**Comment by petermattis** [comment]
LGTM

File: storage/store.go [hunk 3]
```diff
@@ -1581,6 +1589,7 @@ func (s *Store) updateStoreStatus() {
 }
 
 // SetRangeRetryOptions sets the retry options used for this store.
+// For unittests only.
 func (s *Store) SetRangeRetryOptions(ro retry.Options) {
-	s.ctx.RangeRetryOptions = ro
+	s.ctx.RangeRetryOptions = &ro
```
Comment (tschottdorf) [inline storage/store.go]: doesn't this need a lock?

Comment (tschottdorf) [inline storage/store.go]: ok, if it's for unit tests only we can probably ignore that.

**Comment by tschottdorf** [comment]
still LGTM.

**Pull request merged**